the-collab-lab/tcl-12-smart-shopping-list
# Issue 13: Utilized modal to create popup for item details

## Description

- Created "Details" button to view field values of details stored in Firestore database for each item
- Created ItemDetails component to be displayed in our custom modal
- Used media queries to make modal mobile responsive 

## Related Issue

Closes #13

## Acceptance Criteria

- From the shopping list, user can tap to navigate to a detailed view of the item
- Detail view includes the following information:
     - Name of the item
     - Last purchased date
     - Estimated number of days until the next purchase
     - Number of times the item has been purchased
- When in the “item details” view, a back arrow should appear in the header that takes the user back to the “list” view


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

![Screen Shot 2020-11-23 at 8 36 23 PM](https://user-images.githubusercontent.com/61370161/100040004-11ca9700-2dcc-11eb-9202-f221549d6c6c.png)


### After

![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/61370161/100039923-e5af1600-2dcb-11eb-8c88-ced44c69bc04.gif)



## Testing Steps / QA Criteria

- Pull down branch `tc-eb-purchase-details` and start server
- Input mind shoo cuff as a token 
- A "details" button should appear next to the "delete" button. Click this button.
- When you click the button, the modal should appear with: 
    - Name of the item
    - Last purchased date
    - Estimated number of days until the next purchase
    - Number of times the item has been purchased
